pkg/workloadmeta/store.go

```diff
@@ -263,6 +263,24 @@ func (s *store) ListContainers() ([]*Container, error) {
 	return containers, nil
 }
 
+// ListContainersWithFilter implements Store#ListContainersWithFilter
+func (s *store) ListContainersWithFilter(filter ContainerFilterFunc) ([]*Container, error) {
+	allContainers, err := s.ListContainers()
+	if err != nil {
+		return nil, err
+	}
+
+	var res []*Container
+
+	for _, container := range allContainers {
+		if filter(container) {
+			res = append(res, container)
+		}
+	}
+
+	return res, nil
+}
```

Comment:
I think we could make it more performant if we did the filtering in the `ListContainers()` loop [here](https://github.com/DataDog/datadog-agent/blob/3433b169377001bcb4d3e2ed0f86ceb128441e2d/pkg/workloadmeta/store.go#L258-L261)